### ISSUE TITLE
Voltage Data

### ISSUE THREAD

**Issue by flyoffacliff** [magico13/ha-emporia-vue#246]
Although it's still buggy, Emporia recently added voltage monitoring to the official app. Can this integration be updated to pull this value in as another sensor?

![Screenshot_20240123-221052](https://github.com/magico13/ha-emporia-vue/assets/8029878/76e54c35-f343-482c-987b-d1b0b87d1742)

**Comment by magico13** [comment]
I'll look into adding it, though it will increase the number of calls against the API. I may try to add Amps as well at that time since there are requests for that info too. I may keep those disabled by default and make them opt-in, most people probably just want the energy dashboard to work and these extra features are more technical info.

**Comment by Esprit1st** [comment]
"I may keep those disabled by default and make them opt-in" ... Please! Thank you!

**Comment by flyoffacliff** [comment]
Thank you for the quick response! Good to know that the integration is well supported.

I wish they would give clear guidelines on API usage, such as "limit of X calls per account per hour/day".

Hopefully the voltage data is available via API. It appears that the voltage data is stored on their servers as a float with two decimal places, but the app converts it to an int.

I know this because the official app used to display two decimal places in the list view but not the graph view. When I submitted a bug report to them, I asked them to add the decimal places in the graph view, like the other units of measurement had, and instead, they just removed the decimal places from the list view as well. 🙄

**Comment by djripcord** [comment]
> I wish they would give clear guidelines on API usage, such as "limit of X calls per account per hour/day".

I have personally reached out to Emporia about making their API accessible locally. Given their product uses WI-FI, and is able to report data to their servers at _reasonably_ high data rates, there should be no issue implementing a local webserver that simply hosts a JSON string with the same data.

I implore everyone to do the same, and tell them we want local access!

**Comment by EboBH83** [comment]
Really appreciate your work. Looking forward to this addition.

**Comment by ryanbytes** [comment]
Any update on the voltage addition?

**Comment by BenJewell** [comment]
Has anyone even been able to confirm that the voltage data is available over the API at all? I never really found clear API documentation from Emporia.

**Comment by DabblerIOT** [comment]
I'll check the api and see what comes back.   I'd like to have this data as well to monitor my power company for possible brown outs on the very hot days.   Several neighbors have been having some capacitor failures during some of these time periods.   Would be nice to know what voltage was doing.

**Comment by csmracing** [comment]
> **BenJewell:** Has anyone even been able to confirm that the voltage data is available over the API at all?

Yes, you can see the official web app make these calls in developer tools.  Additionally, the PyEmView library which is also developed by the HA Emporia Vue integration author (magico13), supports voltage as of April:

https://github.com/magico13/PyEmVue/blob/master/pyemvue/enums.py



> **magico13:** I'll look into adding it, though it will increase the number of calls against the API. I may try to add Amps as well at that time since there are requests for that info too. 

Just a thought, but you don't have to make three calls to get the information, just two.  If you are already getting watts, just query for amps or volts and then calcuate the 3rd data point.

**Comment by BenJewell** [comment]
I really want this to work and thought it would be easy to add. But the integration is a lot more complex than I thought, and I can't figure out how it calls **pyemvue**.

Since this still doesn't seem to be going anywhere, figured I would give it a try myself, even though I haven't used Python in years. I did this to try and add a toggle for voltage:

<img width="156" alt="image" src="https://github.com/user-attachments/assets/944ed803-2fc7-4c72-b763-dd5ffdb41e2e">

I also created a copy of the 1min coordinator, for volts:

<img width="470" alt="image" src="https://github.com/user-attachments/assets/fa2832b2-26b7-4857-b8c4-707fcbc25231">

But Python throws a **KeyError: 'coordinator_volts'** on this line.

Any ideas? I have a feeling this is not even close.

**Comment by TerribleTesla** [comment]
Any chance this has been looked at in 2025?  It would be ni9ce to be able to capture the voltage fluctuations and track that data in a much nice tool (Home Assistant) instead of the clunky emporia vue app.
